### PR TITLE
Use error string indirection instead of copying.

### PR DESCRIPTION
**Pull request by lzbj**
Try to fix the error message replicate issue, make some change to the MakeWskErrorFromWskError function which could save one copy.

### CODE REVIEW

**Comment by rabbah** [comment]
I'm not sure I follow. Can you show me an example of how to save one of three message identifiers? 

@dubeejw I assume it's  possible to replace the actual message id with a more generic constant identifier. What do you suggest?

**Comment by lzbj** [comment]
Ah, got it. You mean the I18n string replication. Yes, I think we can reference them as tags you mentioned with pointers.

What I changed is the error message passing, now the cost is better, sorry for my misunderstood.

**Comment by dubeejw** [comment]
@mdeuser

**Comment by rabbah** [comment]
I'm not too concerned about the runtime cost in this case - but the developer overhead.

**Comment by lzbj** [comment]
Sure, updated a little. Define the error message as a const as cli convention, if you and everyone agree, I'll continue the remain ones.

File: tools/cli/go-whisk/whisk/action.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func (s *ActionService) List(packageName string, options *ActionListOptions) ([]
     return actions, resp, err
 }
 
-func (s *ActionService) Insert(action *Action, overwrite bool) (*Action, *http.Response, error) {
+func (s *ActionService)         Insert(action *Action, overwrite bool) (*Action, *http.Response, error) {
```
Comment (mdeuser) [inline tools/cli/go-whisk/whisk/action.go]: is the added whitespace intentional?

File: tools/cli/go-whisk/whisk/wskerror.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ const NO_DISPLAY_PREFIX         bool = false
 const APPLICATION_ERR           bool = true
 const NO_APPLICATION_ERR        bool = false
 const TIMED_OUT                 bool = true
+const UNABLE_CREATE_ACTION_ERR  string = "Unable to create action '{{.name}}': {{.err}}"
```
Comment (mdeuser) [inline tools/cli/go-whisk/whisk/wskerror.go]: this change does not seem related to the rest of the commit changes.  this is probably best moved into another pr focusing on using constants as labels for all pii strings

File: tools/cli/go-whisk-cli/commands/action.go [hunk 3]
```diff
@@ -582,50 +582,50 @@ func getLimits(memorySet bool, logSizeSet bool, timeoutSet bool, memory int, log
...
 }
 
-func actionInsertError(action *whisk.Action, err error) (error) {
-    whisk.Debug(whisk.DbgError, "client.Actions.Insert(%#v, false) error: %s\n", action, err)
+func actionInsertError(action *whisk.Action, err *error) (error) {
+    whisk.Debug(whisk.DbgError, "client.Actions.Insert(%#v, false) error: %s\n", action, *err)
 
     errMsg := wski18n.T(
-        "Unable to create action '{{.name}}': {{.err}}",
+        whisk.UNABLE_CREATE_ACTION_ERR,
```
Comment (mdeuser) [inline tools/cli/go-whisk-cli/commands/action.go]: this change does not seem related to the rest of the commit changes.  this is probably best moved into another pr focusing on using constants as labels for all pii strings

**Comment by mdeuser** [comment]
i suggest making the message const changes in another pr

File: tools/cli/go-whisk-cli/commands/action.go (hunk 3 above)
Comment (rabbah) [inline tools/cli/go-whisk-cli/commands/action.go]: @mdeuser is the change of using pointers for the error string worth it?

File: tools/cli/go-whisk/whisk/wskerror.go (hunk 2 above)
Comment (rabbah) [inline tools/cli/go-whisk/whisk/wskerror.go]: @mdeuser this was the spirit of the intended change/this pr. I think @lzbj initially interpreted the issue differently. So we should decide: is it worth making the change of passing the string around by pointer? If so, then yes decouple the prs. If not, drop that commit and making the intended change for reducing the number of times an error message is used. For the latter, I defer to you and @dubeejw and @lzbj for the best way to do that.

File: tools/cli/go-whisk-cli/commands/action.go (hunk 3 above)
Comment (mdeuser) [inline tools/cli/go-whisk-cli/commands/action.go]: as it stands, the conversion to pointer of a subset of error structs is a bit confusing.  in general, the go errors package returns the error struct when creating one, not the pointer to the error struct
https://golang.org/pkg/errors/

Comment (rabbah) [inline tools/cli/go-whisk-cli/commands/action.go]: should we drop this pr then, or repurpose it for the actually intended improvement?

File: tools/cli/go-whisk/whisk/action.go (hunk 1 above)
Comment (lzbj) [inline tools/cli/go-whisk/whisk/action.go]: Np, unintentionally, updated.

**Comment by lzbj** [comment]
@mdeuser, I suggest we keep this pr change, since as you see the MakeWskErrorFromWskError was designed to handle err pointers, the error was in essence is string. Nevertheless, It's up to you decide.

Yes I'll create another PR about the const change.
Anyway, thank you guys for help clarify and review. Appreciate it.

**Comment by mdeuser** [comment]
@lzbj - in general, Go seems to treat errors as stucts and not pointers to structs.  https://golang.org/pkg/errors/  

since WskError is just another implementation of the error interface, i recommend consistency in the treatment of errors - not as pointers, but as structs

**Comment by lzbj** [comment]
Ok, np, close this one. Thanks for your feedback.
